physical/inmem.go

```diff
@@ -27,6 +29,10 @@ func NewInmem() *InmemBackend {
 func (i *InmemBackend) Put(entry *Entry) error {
 	i.l.Lock()
 	defer i.l.Unlock()
+
+	i.permitPool.Acquire()
```

Comment:
I think the permit pool should be acquired before the internal locks to avoid potential deadlock scenario